devito/ir/iet/visitors.py

```diff
@@ -243,10 +259,10 @@ def _gen_value(self, obj, mode=1, masked=()):
                       if getattr(obj.function, k, False) and v not in masked]
 
         if (obj._mem_stack or obj._mem_constant) and mode == 1:
-            strtype = obj._C_typedata
+            strtype = self._complex_type(obj._C_typedata, obj.dtype)
```

Comment:
So all code gen, even for no complex goes through a complex_type check?

Reply:
yeah that was ugly, was mostly for debug , switched to compiler specific types and type lowering